### PR TITLE
G-API: Support CoreML Execution Providers for ONNXRT Backend

### PR DESCRIPTION
**Pull request by TolyaTalamanov**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [ ] I agree to contribute to the project under Apache 2 License.
- [ ] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [ ] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/gapi/include/opencv2/gapi/infer/onnx.hpp [hunk 1]
```diff
@@ -32,6 +32,66 @@ namespace onnx {
  */
 namespace ep {
 
+/**
+ * @brief This structure provides functions
+ * that fill inference options for ONNX OpenVINO Execution Provider.
```
Comment (TolyaTalamanov) [inline modules/gapi/include/opencv2/gapi/infer/onnx.hpp:37]: `CoreML`

File: modules/gapi/include/opencv2/gapi/infer/onnx.hpp [hunk 2]
```diff
@@ -431,6 +494,20 @@ template<typename Net> class Params {
         return *this;
     }
 
+    /** @brief Adds execution provider for runtime.
+
+    The function is used to add ONNX Runtime CoreML Execution Provider options.
+
+    @param ep OpenVINO Execution Provider options.
```
Comment (TolyaTalamanov) [inline modules/gapi/include/opencv2/gapi/infer/onnx.hpp:501]: `CoreML`

File: modules/gapi/include/opencv2/gapi/infer/onnx.hpp (hunk 1 above)
Comment (TolyaTalamanov) [reply modules/gapi/include/opencv2/gapi/infer/onnx.hpp:37]: Done

File: modules/gapi/include/opencv2/gapi/infer/onnx.hpp (hunk 2 above)
Comment (TolyaTalamanov) [reply modules/gapi/include/opencv2/gapi/infer/onnx.hpp:501]: Done

**Comment by dmatveev** [comment]
There's some merge conflicts.

**Comment by asmorkalov** [comment]
@dmatveev please review again.

**Review by dmatveev — APPROVED** [review]
I am ok with the changes but it seems the problem we have now is where we could test this EP work

**Pull request merged**